synthtool/transforms.py

```diff
@@ -213,7 +217,12 @@ def replace(
             log.info(f"Replaced {before!r} in {path}.")
 
     if not any_replaced:
-        log.warning(
-            f"No replacements made in {sources} for pattern {before}, maybe "
-            "replacement is not longer needed?"
-        )
+        if required:
+            log.error(
```

Comment:
@theacodes Will this make the job 'failed' from Autosynth's perspective?

Reply:
I don't think so.